ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractMinerExecutor.java

```diff
@@ -60,10 +68,41 @@ public AbstractMinerExecutor(
     this.gasLimitCalculator = gasLimitCalculator;
   }
 
-  public abstract M startAsyncMining(
-      final Subscribers<MinedBlockObserver> observers, final BlockHeader parentHeader);
+  synchronized void start() {
+    started = true;
+  }
+
+  public M startAsyncMining(
+      final Subscribers<MinedBlockObserver> observers, final BlockHeader parentHeader) {
+    synchronized (this) {
+      assertRunning();
+      final M currentRunningMiner = createMiner(observers, parentHeader);
+      executorService.execute(currentRunningMiner);
+      return currentRunningMiner;
+    }
+  }
+
+  public void stop() {
+    synchronized (this) {
+      if (started && !stopped) {
+        stopped = true;
+      } else {
+        return;
+      }
+    }
+    executorService.shutdownNow();
+  }
+
+  public void awaitStop() {
```

Comment:
It's a general pattern that's used elsewhere in the codebase.  It allows you handle the lifecycle logic more flexibly. 
 For example, you can `stop` a series of components without any waits then follow up with the calls to `awaitStop`.  This speeds up shutdown because we're not waiting for one service to completely stop before we ask the next service to stop.

Reply:
We actually have 2 slightly different patterns.  There's another style of `awaitStop` where we don't use timeouts.  In this case, you can also use the method to wait indefinitely until shutdown.  We should probably make these patterns consistent, and perhaps? create a generic interface to standardize the patterns.  But I won't tackle that in this PR.